external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceProvider.scala

```diff
@@ -353,15 +354,18 @@ private[kafka010] class KafkaSourceProvider extends DataSourceRegister
     }
   }
 
-  class KafkaTable(strategy: => ConsumerStrategy) extends Table
-    with SupportsRead with SupportsWrite {
+  class KafkaTable extends Table with SupportsRead with SupportsWrite {
 
-    override def name(): String = s"Kafka $strategy"
+    override def name(): String = "KafkaTable"
```

Comment:
should we at least put topic name in the kafka table name?

Reply:
Strategy is not always available and this is the case with topic as well.
There is a possibility to provide topic as column (in case of write). Such case the table has multiple topics and only available while rows processed.

Strategy + topic can be added conditionally which most of the time represents the situation.
On the other hand I don't think it worth to parse the rows and update `KafkaTable` name accordingly. Since this provides partial information and the implementation was quite complex compared to the value not sure it has the be added (and that's the reason why not added).

WDYT, should this be added?